### PR TITLE
Convert Atmel SAM0 SPI & UART driver to use dts for sercom pad

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
@mnkp @fallrisk do you guys mind doing a review.

File: dts/bindings/spi/atmel,sam0-spi.yaml [hunk 1]
```diff
@@ -22,4 +22,16 @@ properties:
       description: mmio register space
       generation: define
       category: required
+
+    dipo:
+      type: int
+      category: required
+      description: Receive Data Pinout
```
Comment (mnkp) [inline dts/bindings/spi/atmel,sam0-spi.yaml]: The datasheet refers to this configuration setting as "Data In Pinout".

File: dts/bindings/spi/atmel,sam0-spi.yaml [hunk 2]
```diff
@@ -22,4 +22,16 @@ properties:
...
+      generation: define
+
+    dopo:
+      type: int
+      category: required
+      description: Transmit Data Pinout
```
Comment (mnkp) [inline dts/bindings/spi/atmel,sam0-spi.yaml]: Should be "Data Out Pinout".

**Review by mnkp — CHANGES_REQUESTED** [review]
This looks good. I have only a minor request regarding comments.

On a side note, the way Atmel designed signal routing / pin assignment in SAM0 family is really confusing. Good example of software unfriendly hardware design. But it looks like they were aware of it as the approach taken in SAM family is much, much better.

File: dts/bindings/spi/atmel,sam0-spi.yaml (hunk 2 above)
Comment (galak) [reply dts/bindings/spi/atmel,sam0-spi.yaml]: Yep, copy paste error, good catch.

**Comment by galak** [comment]
@mnkp if you can re-review, made fixes to the spi binding file.

**Review by mnkp — APPROVED** [review]

**Comment by sknsean** [comment]
Breaks UART on atsamd20_xpro :-(

File: boards/arm/atsamd20_xpro/board.h [hunk 3]
```diff
@@ -9,13 +9,4 @@
 
 #include <soc.h>
 
-#define CONFIG_SPI_SAM0_SERCOM0_PADS \
-	(SERCOM_SPI_CTRLA_DIPO(0) | SERCOM_SPI_CTRLA_DOPO(2))
-
-#define CONFIG_UART_SAM0_SERCOM3_PADS \
-	(SERCOM_USART_CTRLA_RXPO_PAD3 | SERCOM_USART_CTRLA_TXPO_PAD2)
```
Comment (sknsean) [inline boards/arm/atsamd20_xpro/board.h]: SERCOM_USART_CTRLA_TXPO_PAD2  = 1

File: boards/arm/atsamd20_xpro/atsamd20_xpro.dts [hunk 4]
```diff
@@ -42,18 +42,24 @@
...
+	dipo = <0>;
+	dopo = <2>;
 };
 
 &sercom3 {
 	status = "ok";
 	compatible = "atmel,sam0-uart";
 	current-speed = <115200>;
+	rxpo = <3>;
+	txpo = <2>;
```
Comment (sknsean) [inline boards/arm/atsamd20_xpro/atsamd20_xpro.dts]: txpo = <1>;

**Pull request merged**